exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/parser/CompoundIdentifierConverter.java

```diff
@@ -107,7 +117,7 @@ public SqlNode visitChild(
       }
       SqlNode newOperand = operand.accept(CompoundIdentifierConverter.this);
       enableComplex = localEnableComplex;
-      if (newOperand != operand) {
+      if (! newOperand.equalsDeep(operand, false)) {
```

Comment:
Is it necessary to call equalsDeep()?  If the expression has an identifier which is rewritten by CompoundIdentifierConverter, is it true that the new expression would be different in reference from then original one?